services/src/main/java/org/keycloak/services/resources/admin/UserResource.java

```diff
@@ -686,9 +686,15 @@ public Response executeActionsEmail(@QueryParam(OIDCLoginProtocol.REDIRECT_URI_P
         }
 
         ClientModel client = realm.getClientByClientId(clientId);
-        if (client == null || !client.isEnabled()) {
+        if (client == null) {
+            logger.debugf("Client %s doesn't exist", clientId);
+            throw new WebApplicationException(
+                ErrorResponse.error("Client doesn't exist", Status.BAD_REQUEST));
+        }
+        if (!client.isEnabled()) {
+            logger.debugf("Client %s is not enabled exist", clientId);
```

Comment:
@slaskawi Thanks! I've just noticed that there seems to be a small typo in the debug message here? Sorry to bother, I hope this one won't take you too much time to fix :)

Reply:
Fixed. Thanks for spotting this!